docs/jwst/user_documentation/running_pipeline_python.rst

```diff
@@ -529,3 +529,196 @@ individual step parameter must be set when using this method, or else the coded
 defaults will be used, which may be inappropriate for the dataset being processed.
 
 See :ref:`call_examples` for more information.
+
+
+.. _multiprocessing:
+
+Multiprocessing
+===============
+
+Python's multiprocessing module explicitly imports and executes a script's
+`__main__` with each and every worker. If `__main__` is not present the behavior is
+undefined. Hence, Python will crash unless the multiprocess code in enclosed in a
+`__main__` block like this:
+
+
+::
+
+    import sys
+
+    def main():
+        [code used in multiprocessing]
+
+    if __name__ = '__main__':
+        sys.exit(main())
+
+
+There are three scenarios to use multiprocessing with the pipeline:
+
+1. Multiprocessing with a pipeline step. At the moment, these steps are
+:ref:`jump_step <jump_step>`, :ref:`ramp_fitting_step <ramp_fitting_step>`,
+and :ref:`wfss_contam_step <wfss_contam_step>`. To enable multiprocessing the
+optional parameters are `max_cores` for the ``jump`` step, and `maximum_cores`
+for the ``ramp_fitting`` and ``wfss_contam`` steps. These parameters can be
+set to `quarter`, `half`, `all`, or `none`, which is the default value.
+
+The following example turns on the step multiprocessing while setting up a log
+file for each run of the pipeline and a text file with the full traceback in case
+there is a crash. Notice only one of the steps has multiprocessing turned on. We
+do not recommend to simultaneously enable both steps to do multiprocessing, as
+this may likely lead to running out of system memory.
+
+
+
+::
+
+    # SampleScript1
+
+    import os, sys
+    import traceback
+    import configparser
+    from glob import glob
+    from jwst.pipeline import Detector1Pipeline
+
+    files = glob('*uncal.fits')
+    output_dir = '/my_project'
+
+    def mk_stpipe_log_cfg(output_dir, log_name):
+        """
+        Create a configuration file with the name log_name, where
+        the pipeline will write all output.
+        Args:
+            outpur_dir: str, path of the output directory
+            log_name: str, name of the log to record screen output
+        Returns:
+            nothing
+        """
+        config = configparser.ConfigParser()
+        config.add_section("*")
+        config.set("*", "handler", "file:" + log_name)
+        config.set("*", "level", "INFO")
+        pipe_log_config = os.path.join(output_dir, "pipeline-log.cfg")
+        config.write(open(pipe_log_config, "w"))
+
+    def main():
+        for item in files:
```

Comment:
I would suggest making this example simpler by running it over only a single file, since what is being highlighted here is now to multiprocess on one file within a step

Reply:
I reduced the code to highlight the multiprocessing part.